### PR TITLE
Feature/test notification

### PR DESCRIPTION
**Pull request by zelick**

### CODE REVIEW

File: Konteh/Konteh.BackOfficeApi/Program.cs [hunk 1]
```diff
@@ -44,26 +51,37 @@ private static void Main(string[] args)
                 {
                     builder.WithOrigins("http://localhost:4200")
                            .AllowAnyMethod()
-                           .AllowAnyHeader();
+                           .AllowAnyHeader()
```
Comment (Djoleoo) [inline Konteh/Konteh.BackOfficeApi/Program.cs:54]: are you sure you need these

File: Konteh/Konteh.BackOfficeApi/Properties/launchSettings.json [hunk 2]
```diff
@@ -22,14 +22,14 @@
     "https": {
       "commandName": "Project",
       "dotnetRunMessages": true,
-      "launchBrowser": true,
+      "launchBrowser": true, 
       "launchUrl": "swagger",
```
Comment (Djoleoo) [inline Konteh/Konteh.BackOfficeApi/Properties/launchSettings.json:26]: weird change but ok

File: Konteh/back-office-app/.angular/cache/18.2.3/vite/deps/_metadata.json [hunk 3]
```diff
@@ -13,7 +13,7 @@
     "@angular/common/http": {
       "src": "../../../../../node_modules/@angular/common/fesm2022/http.mjs",
```
Comment (Djoleoo) [inline Konteh/back-office-app/.angular/cache/18.2.3/vite/deps/_metadata.json:14]: ???

File: .gitignore [hunk 4]
```diff
@@ -383,3 +383,5 @@ _metadata.json
 .angular/cache/18.2.3
 
 
+*.json
```
Comment (laslorajda) [inline .gitignore:386]: I don't think we can just ignore json files, they contain a lot of configurations.

File: Konteh/Konteh.BackOfficeApi/Consumers/ExamRequestedConsumer.cs [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+public class ExamRequestedConsumer : IConsumer<ExamRequestedEvent>
+{
+    private readonly IHubContext<ExamHub> _hubContext;
+    public ExamRequestedConsumer(IHubContext<ExamHub> hubContext)
+    {
+        _hubContext = hubContext;
+    }
+    public async Task Consume(ConsumeContext<ExamRequestedEvent> context)
+    {
+        var jsonMessage = JsonSerializer.Serialize(context.Message);
```
Comment (laslorajda) [inline Konteh/Konteh.BackOfficeApi/Consumers/ExamRequestedConsumer.cs:18]: I don't think we need to serialize our objects, SignalR should be able to do it for us.

File: Konteh/Konteh.BackOfficeApi/Consumers/ExamRequestedConsumer.cs [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+        await _hubContext.Clients.All.SendAsync("ReceiveExamRequest", jsonMessage);
+        await Task.CompletedTask;
```
Comment (laslorajda) [inline Konteh/Konteh.BackOfficeApi/Consumers/ExamRequestedConsumer.cs:20]: I don't see the point of awaiting this task

File: Konteh/Konteh.BackOfficeApi/Program.cs [hunk 7]
```diff
@@ -44,26 +51,37 @@ private static void Main(string[] args)
...
+            x.AddConsumer<ExamRequestedConsumer>();
+
+            x.UsingRabbitMq((context, cfg) =>
+            {
+                cfg.Host("localhost", "/", h =>
                 {
-                    builder.WithOrigins("http://localhost:4200")
-                           .AllowAnyMethod()
-                           .AllowAnyHeader();
+                    h.Username("admin");
```
Comment (laslorajda) [inline Konteh/Konteh.BackOfficeApi/Program.cs:68]: all these things should probably be parametrized. Look into the options pattern.

File: Konteh/Konteh.FrontOfficeApi/Program.cs [hunk 8]
```diff
@@ -24,6 +25,20 @@ private static void Main(string[] args)
...
         builder.Services.AddMediatR(cfg => cfg.RegisterServicesFromAssembly(Assembly.GetExecutingAssembly()));
 
+
+        builder.Services.AddMassTransit(x =>
+        {
+            x.UsingRabbitMq((context, cfg) =>
+            {
+                cfg.Host("localhost", "/", c =>
+                {
+                    c.Username("admin");
```
Comment (laslorajda) [inline Konteh/Konteh.FrontOfficeApi/Program.cs:35]: same here

File: Konteh/back-office-app/src/app/services/signar-service.ts [hunk 9]
```diff
@@ -0,0 +1,26 @@
...
+  private hubConnection!: signalR.HubConnection;
+
+  startConnection(hubUrl: string): void {
+    this.hubConnection = new signalR.HubConnectionBuilder()
+    .withUrl(hubUrl)
+    .build();
+
+    this.hubConnection
+    .start()
+    .then(() => console.log('Connection started'))
```
Comment (laslorajda) [inline Konteh/back-office-app/src/app/services/signar-service.ts:17]: no console logs in repo please

File: Konteh/Konteh.FrontOfficeApi/Program.cs [hunk 10]
```diff
@@ -37,14 +52,24 @@ private static void Main(string[] args)
                 });
         });
 
+        builder.Services.AddCors(options =>
+        {
+            options.AddPolicy("AllowSpecificOrigins",
+                builder =>
+                {
+                    builder.WithOrigins("http://localhost:5000")
```
Comment (laslorajda) [inline Konteh/Konteh.FrontOfficeApi/Program.cs:60]: how do we know the application will run on port 5000?I see no guarantees of that

**Review by laslorajda — APPROVED** [review]

**Pull request merged**